<a href="https://colab.research.google.com/github/izzat-ai/learning-ai/blob/main/projects/movie.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [17]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [18]:
df = pd.read_csv("/content/Top_rated_movies.csv")
df.head()

,id,title,original_language,original_title,overview,release_date,popularity,vote_count,rating
0,13448,Angels & Demons,en,Angels & Demons,Harvard symbologist Robert Langdon is recruite...,2009-04-20,6.8861,7323,6.719
1,9603,Clueless,en,Clueless,"Shallow, rich and socially successful Cher is ...",1995-07-19,7.5190,4801,7.255
2,6557,27 Dresses,en,27 Dresses,Altruistic Jane finds herself facing her worst...,2008-01-10,5.7786,3421,6.453
3,1333100,Attack on Titan: THE LAST ATTACK,ja,劇場版「進撃の巨人」完結編 THE LAST ATTACK,A colossal-sized omnibus film bringing togethe...,2024-03-20,6.0979,205,8.700
4,578701,Those Who Wish Me Dead,en,Those Who Wish Me Dead,A young boy finds himself pursued by two assas...,2021-05-05,6.2045,1849,6.607


In [19]:
df2 = pd.read_csv("/content/movie_data.csv")
df2.head()

,id,title,original_language,overview,release_date,original_title,vote_average,vote_count
0,1339713,Obsession,en,"After breaking the mysterious ""One Wish Willow...",2026-05-13,Obsession,7.900,497
1,1304313,Lee Cronin's The Mummy,en,The young daughter of a journalist disappears ...,2026-04-15,Lee Cronin's The Mummy,8.056,1548
2,1433117,Kara,ta,"A thief tries to go straight, but when predato...",2026-04-30,கர,5.800,14
3,1380291,Tom Clancy's Jack Ryan: Ghost War,en,Jack Ryan is reluctantly pulled back into espi...,2026-05-20,Tom Clancy's Jack Ryan: Ghost War,7.105,457
4,1083381,Backrooms,en,A strange doorway appears in the basement of a...,2026-05-27,Backrooms,6.700,275


##EDA

In [20]:
df.columns

Index(['id', 'title', 'original_language', 'original_title', 'overview',
       'release_date', 'popularity', 'vote_count', 'rating'],
      dtype='object')

In [21]:
df.shape

(10000, 9)

In [22]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 10000 non-null  int64  
 1   title              10000 non-null  object 
 2   original_language  10000 non-null  object 
 3   original_title     10000 non-null  object 
 4   overview           9500 non-null   object 
 5   release_date       10000 non-null  object 
 6   popularity         10000 non-null  float64
 7   vote_count         10000 non-null  int64  
 8   rating             10000 non-null  float64
dtypes: float64(2), int64(2), object(5)
memory usage: 703.3+ KB


1. overview ustunida 500 ta ma'lumotlar yo'q
2. release_date ustuni object turida

In [23]:
df.describe()

,id,popularity,vote_count,rating
count,1.000000e+04,10000.000000,10000.000000,10000.000000
mean,6.165003e+05,6.433940,2765.800000,6.210350
std,6.230573e+05,0.547213,3236.140405,2.335112
min,9.750000e+02,5.473400,0.000000,0.000000
25%,9.526500e+03,6.087075,160.000000,5.960500
50%,5.127105e+05,6.466050,1383.500000,6.714500
75%,1.186288e+06,6.879800,4263.500000,7.487750
max,1.653208e+06,7.519000,11993.000000,8.700000


In [24]:
df.duplicated().sum()

np.int64(9980)

In [25]:
print(df.duplicated().sum())

9980


In [26]:
df.sample(5)

,id,title,original_language,original_title,overview,release_date,popularity,vote_count,rating
3198,1137350,The Phoenician Scheme,en,The Phoenician Scheme,Wealthy businessman Zsa-zsa Korda appoints his...,2025-05-23,6.0546,918,6.500
3442,6557,27 Dresses,en,27 Dresses,Altruistic Jane finds herself facing her worst...,2008-01-10,5.7786,3421,6.453
3524,578701,Those Who Wish Me Dead,en,Those Who Wish Me Dead,A young boy finds himself pursued by two assas...,2021-05-05,6.2045,1849,6.607
7377,516486,Greyhound,en,Greyhound,A first-time captain leads a convoy of allied ...,2020-07-09,7.0485,3313,7.417
1603,1333100,Attack on Titan: THE LAST ATTACK,ja,劇場版「進撃の巨人」完結編 THE LAST ATTACK,A colossal-sized omnibus film bringing togethe...,2024-03-20,6.0979,205,8.700


- Yuqorida duplikatlarni aniqladik , ular 9980 ta chiqdi ya'ni takrorlangan ma'lumotlar . Kaggle'dagi bu dataset API orqali yuklanayotganda yoki generatsiya qilinayotganda texnik xatolik tufayli aynan bir xil qatorlar qayta-qayta (yuzlab marta) nusxalanib, pastga qo'shilib ketgan
- Bu muammoni Data Cleaningda tozalab ketamiz

In [27]:
df['original_language'].value_counts()

,count
original_language,
en,7000
zh,1000
ja,500
tl,500
cn,500
hi,500


In [30]:
df['title'].value_counts()

,count
title,
Angels & Demons,500
Clueless,500
27 Dresses,500
Attack on Titan: THE LAST ATTACK,500
Those Who Wish Me Dead,500
Kong Tao,500
The Bourne Supremacy,500
Monster House,500
The Eight Hundred,500


- Juda ko'p ma'lumotlar takrorlangan

##Data Cleaning

In [28]:
df_copy = df.drop_duplicates()
df_copy.shape

(20, 9)

In [31]:
df_copy.reset_index()

,index,id,title,original_language,original_title,overview,release_date,popularity,vote_count,rating
0,0,13448,Angels & Demons,en,Angels & Demons,Harvard symbologist Robert Langdon is recruite...,2009-04-20,6.8861,7323,6.719
1,1,9603,Clueless,en,Clueless,"Shallow, rich and socially successful Cher is ...",1995-07-19,7.5190,4801,7.255
2,2,6557,27 Dresses,en,27 Dresses,Altruistic Jane finds herself facing her worst...,2008-01-10,5.7786,3421,6.453
3,3,1333100,Attack on Titan: THE LAST ATTACK,ja,劇場版「進撃の巨人」完結編 THE LAST ATTACK,A colossal-sized omnibus film bringing togethe...,2024-03-20,6.0979,205,8.700
4,4,578701,Those Who Wish Me Dead,en,Those Who Wish Me Dead,A young boy finds himself pursued by two assas...,2021-05-05,6.2045,1849,6.607
5,5,1529513,Kong Tao,zh,蛊降,Local property developer Song Longchen secretl...,2026-03-19,5.6685,2,8.500
6,6,2502,The Bourne Supremacy,en,The Bourne Supremacy,A CIA operation to purchase classified Russian...,2004-07-23,6.6624,8158,7.332
7,7,9297,Monster House,en,Monster House,"Monsters under the bed are scary enough, but w...",2006-06-30,6.2944,4328,6.710
8,8,508935,The Eight Hundred,zh,八佰,"In 1937, eight hundred Chinese soldiers fight ...",2020-08-14,5.5712,336,7.000
9,9,1576198,Madulas,en,Madulas,"At the luxurious Wet & Wild Resort, a group of...",2025-11-14,6.4494,1,4.000


- Modelimiz sun'iy ravishda ko'paytirilgan bir xil ma'lumotlarni qayta-qayta ko'rib, "yodlab" olmasligi va haqiqiy qonuniyatlarni o'rganishi uchun faqat toza va noyob qatorlar bilan ishlashimiz shart
- Yuqorida esa atiga 20 ta qator va 9 ta ustunli ma'lumotlar qoldi . Bu Machine Learning modeli qurish uchun mutlaqo yetarsiz
- Demak , ikkinchi faylni tekshiramiz !

##Ikkinchi datasetni tekshirish . EDA

In [32]:
df2.head()

,id,title,original_language,overview,release_date,original_title,vote_average,vote_count
0,1339713,Obsession,en,"After breaking the mysterious ""One Wish Willow...",2026-05-13,Obsession,7.900,497
1,1304313,Lee Cronin's The Mummy,en,The young daughter of a journalist disappears ...,2026-04-15,Lee Cronin's The Mummy,8.056,1548
2,1433117,Kara,ta,"A thief tries to go straight, but when predato...",2026-04-30,கர,5.800,14
3,1380291,Tom Clancy's Jack Ryan: Ghost War,en,Jack Ryan is reluctantly pulled back into espi...,2026-05-20,Tom Clancy's Jack Ryan: Ghost War,7.105,457
4,1083381,Backrooms,en,A strange doorway appears in the basement of a...,2026-05-27,Backrooms,6.700,275


In [33]:
df2.shape

(2000, 8)

In [34]:
df2.columns

Index(['id', 'title', 'original_language', 'overview', 'release_date',
       'original_title', 'vote_average', 'vote_count'],
      dtype='object')

In [35]:
df2.duplicated().sum()

np.int64(8)

In [37]:
# asl dataframedan nusxa olish
df_copy = df2.copy()
df_copy

,id,title,original_language,overview,release_date,original_title,vote_average,vote_count
0,1339713,Obsession,en,"After breaking the mysterious ""One Wish Willow...",2026-05-13,Obsession,7.900,497
1,1304313,Lee Cronin's The Mummy,en,The young daughter of a journalist disappears ...,2026-04-15,Lee Cronin's The Mummy,8.056,1548
2,1433117,Kara,ta,"A thief tries to go straight, but when predato...",2026-04-30,கர,5.800,14
3,1380291,Tom Clancy's Jack Ryan: Ghost War,en,Jack Ryan is reluctantly pulled back into espi...,2026-05-20,Tom Clancy's Jack Ryan: Ghost War,7.105,457
4,1083381,Backrooms,en,A strange doorway appears in the basement of a...,2026-05-27,Backrooms,6.700,275
...,...,...,...,...,...,...,...,...
1995,36955,True Lies,en,"A fearless, globe-trotting, terrorist-battling...",1994-07-15,True Lies,7.096,4575
1996,36669,Die Another Day,en,James Bond is sent to investigate the connecti...,2002-11-17,Die Another Day,5.998,3885
1997,1270125,Monster on a Plane,en,An exotic animal turns into a monster and beco...,2024-11-07,Monster on a Plane,4.778,45
1998,395000,Young Mother 4,ko,"Woo-jin, who is a repeater of college entrance...",2016-04-28,젊은 엄마 4,4.500,12


In [42]:
# duplikatlarni o'chirish
df_copy = df_copy.drop_duplicates()
df_copy.shape

(1992, 8)

In [44]:
# df ning birinchi ustunini indeks qilish
df_copy = df_copy.set_index(df_copy.columns[0]).reset_index()
df_copy

,id,title,original_language,overview,release_date,original_title,vote_average,vote_count
0,1339713,Obsession,en,"After breaking the mysterious ""One Wish Willow...",2026-05-13,Obsession,7.900,497
1,1304313,Lee Cronin's The Mummy,en,The young daughter of a journalist disappears ...,2026-04-15,Lee Cronin's The Mummy,8.056,1548
2,1433117,Kara,ta,"A thief tries to go straight, but when predato...",2026-04-30,கர,5.800,14
3,1380291,Tom Clancy's Jack Ryan: Ghost War,en,Jack Ryan is reluctantly pulled back into espi...,2026-05-20,Tom Clancy's Jack Ryan: Ghost War,7.105,457
4,1083381,Backrooms,en,A strange doorway appears in the basement of a...,2026-05-27,Backrooms,6.700,275
...,...,...,...,...,...,...,...,...
1987,36955,True Lies,en,"A fearless, globe-trotting, terrorist-battling...",1994-07-15,True Lies,7.096,4575
1988,36669,Die Another Day,en,James Bond is sent to investigate the connecti...,2002-11-17,Die Another Day,5.998,3885
1989,1270125,Monster on a Plane,en,An exotic animal turns into a monster and beco...,2024-11-07,Monster on a Plane,4.778,45
1990,395000,Young Mother 4,ko,"Woo-jin, who is a repeater of college entrance...",2016-04-28,젊은 엄마 4,4.500,12


## Data Cleaning

In [45]:
df_copy.isnull().sum()

,0
id,0
title,0
original_language,0
overview,15
release_date,4
original_title,0
vote_average,0
vote_count,0
